acceptance/acceptance_test.go

```diff
@@ -346,6 +347,7 @@ func testWithoutSpecificBuilderRequirement(
 						"package-buildpack", packageName,
 						"-c", aggregatePackageToml,
 						"--publish",
+						"--os", dockerHostOS(),
```

Comment:
I wonder if there's a way to infer this without a flag... I understand why we might not want to look at the dockerhost OS, as you might be cross-building. Though wouldn't that only work in a `--publish` scenario?

Back to the original question, is there some way we could derive this from the buildpack? Seems unlikely but thought I'd ask.

Reply:
Yeah, that's a good thing to discuss. Right now, for backward compatibility reasons, this implementation does still always default to `linux` for `--publish` or `--format=file` w/o the `--os` flag (it emits a warning telling you this) but for local images (`package-buildpack` w/o options) it will autodetect the image type (without a warning) and save it on the dameon. Also for backward compatibility, it can't _require_ a daemon for `--publish` or `--format=file`. 

With that in mind, the autodetection options for `--publish` or `--format=file` might be to detect based on a _optional_ daemon, but depending on if your daemon is running/stopped, you have a reachable/unreachable `DOCKER_HOST` set, or some other environment difference, you'd generate different images with the same command.

Moreover, autodetection might get surprising for new users - especially on a Windows workstation, which depending on whether you picked LCOW or WCOW, would make different remote images and `.cnb` files for the same command.

I feel like autodetecting the image type only really feels right in the case of a local image on the daemon, where it has to match the running daemon anyways and it unlikely to be pushed up to a registry or copied around.

I'd love to hear any holes in that logic though

Reply:
We *could* make the `os` an optional field in the package config though, instead of the `--os` flag. Then `package-buildpack` would consistently generate the same image time regardless of daemon. The only potential catch there is that to-date, the package config tends to describe CNB-related things (buildpack paths and images) inside a package and not things about the way it's packaged (ex: `--format` is specified on the command line, not in the `package.toml`).